sale_delivery_date/models/sale_order_line.py

```diff
@@ -87,11 +89,38 @@ def _deduce_workload_and_security_lead(
         account the workload, the WH calendar and the cutoff time.
         """
         # Remove the security lead from the date_deadline
+        # If there's a calendar, remove 1 day until we get to a working day.
+        # The reason for that is the parcel cannot be given to the carrier
+        # on a non-working day.
+        calendar = warehouse.calendar_id
         date_planned = date_deadline - timedelta(days=security_lead)
-        # Deduce the workload
-        date_planned = self._apply_workload(date_planned, -days, warehouse.calendar_id)
-        # the correct day has already been computed, only change the cut-off time
-        return self._apply_cutoff(date_planned, partner, warehouse, keep_same_day=True)
+        if calendar:
+            next_working_day = self._next_working_day(date_planned, calendar)
+            count = 0
+            while next_working_day.date() != date_planned.date():
+                if count > FIND_WORKING_DAY_COUNT:  # To avoid infinite loop
+                    raise exceptions.UserError(
+                        _(
+                            "Unable to find a working day matching "
+                            "customer's delivery time window."
+                        )
+                    )
+                date_planned -= timedelta(days=1)
+                next_working_day = self._next_working_day(date_planned, calendar)
+                count += 1
```

Comment:
This part is almost duplicated from `_apply_delivery_window`, could be cool to put the common part in a method, not blocking.

Reply:
My thought exactly.